rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -1357,10 +1357,11 @@ bool PlayerImpl::publish_message(rosbag2_storage::SerializedBagMessageSharedPtr
 {
   auto pub_iter = publishers_.find(message->topic_name);
   if (pub_iter != publishers_.end()) {
-    // Calling on play message pre-callbacks
-    run_play_msg_pre_callbacks(message);
     bool message_published = false;
     try {
+      // Calling on play message pre-callbacks
+      run_play_msg_pre_callbacks(message);
```

Comment:
This should be in its own `try`/`catch` and log a `Failed to call on play message pre-callback on ...` error, otherwise an exception in `run_play_msg_pre_callbacks()` will be reported as a publish failure.

Reply:
The idea was in a specific logic:
if the preconditions in terms of the pre-callbacks are not met, i.e. if an exception is thrown in one of the pre-callbacks,  then the publish method shouldn't be called.  
As far as I recall, we rely on this logic in one of our downstream packages, which uses the player directly.

Reply:
Good point, but this logic could still be achieved using separate `try`/`catch` blocks (e.g., using a simple `bool`) so that each one has its own error message. Something like this:

```cpp
bool do_publish = false;
try {
  run_play_msg_pre_callbacks(message);
  do_publish = true;
} catch (e) {
  // Log pre-callback error
}

if (do_publish) {
    try {
      // publish()
    } catch (e) {
      // Log publish error
  }
}
```

It has the same behaviour (skips publishing if pre-callback throws) but logs a helpful error message that directly mentions that a pre-callback threw.

Reply:
Okay, I will rewrite it this way and will add a unit test for the case when `play_next` returns false if `pre_callback` throws.